### DISCUSSION TITLE
Copying downloads folder... so fragmented?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#22981] [Q&A]
Setting up a build on a new machine with no internet and very low-speed network, I need to install vcpkg. I can get the git repo across easily enough and I know I can copy the vcpkg\downloads folder... but wow it's messy.

A quick check of mine has a total size of 1.8GB. About half that is across 234 files in the download root (gz files, zips, etc) but the other half is in my downloads\tools folder, smeared over 42 _thousand_ files. 

Surely this isn't the raw download vcpkg itself downloads when run for the first time? It must be the result of de-archiving some other files, I assume... any tips how best to do it because copying tens of thousands of small files isn't great! Neither is zipping it all up, that's going to take some time!

Do I _need_ the tools sub-folder or is this all built from other download files?

**Comment by autoantwort** [comment]
> Do I need the tools sub-folder or is this all built from other download files?

Yes, otherwise the tools are downloaded again

**Reply by jdx-john** [reply]
> > Do I need the tools sub-folder or is this all built from other download files?
> 
> Yes, otherwise the tools are downloaded again

I'm not sure about this. I tested copying downloads without _tools_ sub-dir and trying to install, and I saw it extracted/installed portable powershell, nuget, cmake as normal even though there was no internet connection. Seems a bit weird.

**Reply by autoantwort** [reply]
Then it uses the already downloaded stuff from `downloads`

**Reply by jdx-john** [reply]
this definitely makes life easier!

**Comment by autoantwort** [comment]
But if you have intranet, simply use asset caching: https://github.com/microsoft/vcpkg/discussions/20409#discussioncomment-1395827
